syntax/scan.go

```diff
@@ -762,34 +804,62 @@ func (sc *scanner) scanString(val *tokenValue, quote rune) Token {
 	if triple {
 		sc.readRune()
 		sc.readRune()
-	}
 
-	quoteCount := 0
-	for {
-		if sc.eof() {
-			sc.error(val.pos, "unexpected EOF in string")
-		}
-		c := sc.readRune()
-		if c == '\n' && !triple {
-			sc.error(val.pos, "unexpected newline in string")
-		}
-		if c == quote {
-			quoteCount++
-			if !triple || quoteCount == 3 {
-				break
+		// A triple-quoted string literal may span multiple
+		// gulps of REPL input; it is the only such token.
+		// Thus we must avoid {start,end}Token.
+		var raw bytes.Buffer
+
+		// Copy the prefix, e.g. r''' or """ (see startToken).
+		raw.Write(sc.token[:len(sc.token)-len(sc.rest)])
+
+		quoteCount := 0
+		for {
+			if sc.eof() {
+				sc.error(val.pos, "unexpected EOF in string")
+			}
+			c := sc.readRune()
+			raw.WriteRune(c)
+			if c == quote {
+				quoteCount++
+				if quoteCount == 3 {
+					break
+				}
+			} else {
+				quoteCount = 0
+			}
+			if c == '\\' {
```

Comment:
Seems reasonable to me: the backslash escapes the first quote, then the following two are not enough to make a close quote.

Reply:
Python raw string literals allow you to escape only a quote, unlike Go backquote literals which are truly raw. You could argue that it's inconsistent, but I imagine it might be a pragmatic choice. It also simplifies this logic here, which doesn't need to care whether it's raw or not when handling backslash.